### PR TITLE
20180206 bulk category

### PR DESCRIPTION
**Pull request by mannyrivera2010**
AMLNG-534

**Description**     
As a developer, I want to be able to access an API so that I can create calls to update multiple listings' categories at once.

**Acceptance Criteria**   
 - A 'GET' endpoint is created that retrieves listings filtered by Category 
 - Filtered Dynamically by role 
If the Role of the user = Admin then show all
If the Role of the user = Org-steward then show listing under that Org  
If the role of the user = regular user then show listing that user owns
 - A 'PUT' endpoint is created that updates multiple listing's category values at once.
 
**How to test code**    
GET Endpoint
```
GET /api/category/7/listing/ HTTP/1.1
Host: 127.0.0.1:8001
Authorization: Basic am9uZXM6cGFzc3dvcmQ=    # jones

[
    {
        "unique_name": "applied_ethics_inc.",
        "title": "Applied Ethics Inc.",
        "id": 7,
        "agency": {
            "title": "Ministry of Plenty",
            "short_name": "Miniplen"
        },
        "categories": [
            {
                "title": "Business",
                "description": "For making money"
            },
            {
                "title": "Finance",
                "description": "For managing money"
            }
        ],
        "owners": [
            {
                "id": 5,
                "user": {
                    "username": "wsmith"
                },
                "display_name": "Winston Smith"
            },
            {
                "id": 13,
                "user": {
                    "username": "jones"
                },
                "display_name": "Jones"
            }
        ]
    },.....
]
```
Update Endpoint
```
POST /api/category/7/listing/ HTTP/1.1
Host: 127.0.0.1:8001
Authorization: Basic am9uZXM6cGFzc3dvcmQ=    # jones
Content-Type: application/json

[
    {
        "id": 7,
        "categories": [
            {
                "title": "Business",
                "description": "For making money"
            },
            {
                "title": "Communication",
                "description": "For managing money"
            }
        ]
    },
    {
        "id": 67,
        "categories": [
         {
                "title": "Business",
                "description": "For making money"
            },
            {
                "title": "Communication",
                "description": "Moving info between people and things"
            }
        ]
    }
]
``` 
Run Test
```
make test_parallel
```

**Please Review**     
@aml-development/ozp-backend-team    

**Checklist**
- [x] Read CONTRIBUTING.md
- [x] Write code to complete task 
- [x] Python Code is PEP8 Compliant
- [x] Add unit tests for new code
- [x] All tests must pass before merging the pull request
- [ ] At least 2 people reviewed code

### CODE REVIEW

File: ozpcenter/models.py [hunk 1]
```diff
@@ -1021,6 +1021,7 @@ def apply_select_related(self, queryset):
 
     def get_queryset(self):
         queryset = super(AccessControlListingManager, self).get_queryset()
+        return queryset
```
Comment (mannyrivera2010) [inline ozpcenter/models.py]: remove this line

File: ozpcenter/api/category/serializers.py [hunk 2]
```diff
@@ -27,3 +33,136 @@ class Meta:
...
+
+    def create(self, validated_data):
+        listing_object = validated_data['listing']
+        listing_object.categories.clear()
+
+        for category_object in validated_data['categories']:
+            print(category_object)
+            listing_object.categories.add(category_object)
+
+        print('-' * 20)
```
Comment (mannyrivera2010) [inline ozpcenter/api/category/serializers.py]: remove this line

**Comment by mannyrivera2010** [comment]
:+1:

**Comment by mannyrivera2010** [comment]
```
curl -X POST \
  http://127.0.0.1:8001/api/category/7/listing/ \
  -H 'authorization: Basic am9uZXM6cGFzc3dvcmQ=' \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json' \
  -H 'postman-token: 8a9554ee-c78d-671d-22a3-9c4c5adf8937' \
  -d '[
    {
        "id": 7,
        "categories": [
            {
                "title": "Business",
                "description": "For making money"
            },
            {
                "title": "Communication",
                "description": "For managing money"
            }
        ]
    },
    {
        "id": 67,
        "categories": [
         {
                "title": "Business",
                "description": "For making money"
            },
            {
                "title": "Communication",
                "description": "Moving info between people and things"
            }
        ]
    }
]'
```

**Comment by mleeBoeing** [comment]
:+1:

**Comment by mleeBoeing** [comment]
Is there a test that covers the scenario when a user who doesn't have the correct permission tries to update?

**Pull request merged**